### PR TITLE
[WIP] docs: Add glossary

### LINKED ISSUE #13532: Add a glossary to docs
### description

The Envoy docs (and community discussions) use a lot of terminology that is not always clear/precise to new users

Some phrases are Envoy-specific, others are common in related communities (eg serverless, service mesh, etc). Many use acronyms

It would be good to have a glossary so key terms used in the Envoy docs can be set out clearly

Terms that immediately spring to mind

- `typed_config`
- [a-Z]DS
- middle/edge proxy
- data plane
- control plane
- circuit breaker
- (Envoy) filter
- (Envoy) extension
- (Envoy) Bootstrap
- sidecar

im happy to collate/write the glossary, but suggestions of terms and/or definitions would be greatly appreciated

### PR DESCRIPTION
**Pull request by phlax**
Signed-off-by: Ryan Northey <ryan@synca.io>

Commit Message: docs: Add glossary
Additional Description:
Risk Level:
Testing:
Docs Changes:
Release Notes:
Platform Specific Features:
[Optional Runtime guard:]
[Optional Fixes #Issue] Fix #13532 
[Optional Deprecated:]

### CODE REVIEW

**Comment by phlax** [comment]
rendered docs are here https://storage.googleapis.com/envoy-pr/13829/docs/glossary.html

**Comment by mattklein123** [comment]
Awesome. Reminder that we should merge somehow with https://www.envoyproxy.io/docs/envoy/latest/intro/arch_overview/intro/terminology.

/wait

File: .azure-pipelines/pipelines.yml [hunk 1]
```diff
@@ -59,7 +59,10 @@ stages:
               GCP_SERVICE_ACCOUNT_KEY: $(GcpServiceAccountKey)
             displayName: "Generate docs"
 
-          - script: ci/run_envoy_docker.sh 'ci/upload_gcs_artifact.sh /source/generated/docs docs'
+          - script: |
+              SHORT_COMMIT_SHA=$(git log --pretty=%P -n 1 | cut -d' ' -f2 | head -c7)
```
Comment (phlax) [inline .azure-pipelines/pipelines.yml:63]: revert workaround

**Comment by phlax** [comment]
> Awesome. Reminder that we should merge somehow with...

there is also this:

https://www.envoyproxy.io/docs/envoy/latest/intro/life_of_a_request#terminology

im thinking we should consolidate that too

**Comment by phlax** [comment]
bump

**Comment by phlax** [comment]
bump

**Comment by phlax** [comment]
bump

**Pull request closed without merging**